clang/include/clang/Basic/Attr.td

```diff
@@ -1135,6 +1135,20 @@ def SYCLKernel : InheritableAttr {
   let Documentation = [SYCLKernelDocs];
 }
 
+def SYCLSpecialClass: InheritableAttr {
```

Comment:
FYI: @erichkeane recently added `SYCLRequiresDecomposition` attribute, which probably can be re-used instead of `SYCLSpecialClass`.

Reply:
`SYCLRequiresDecomposition` attribute appears on user-defined struct kernel arguments if they should be passed to the kernel field-by-field, then these structs are constructed back on kernel side without `init` method usage. Classes which we want to identify with `SYCLSpecialClass` (accessor, sampler, stream) are handled specially with `init` method. So, technically right now `SYCLRequiresDecomposition` attribute works in a bit other use case. To re-use this attribute we will need to change some logic.

Reply:
presumably SYCLSpecialClass is a subset of the SYCLRequiresDecomposition.  I think we still would need BOTH, as the Special-class needs to understand how to deal with the 'init' function, but it should also be translated to the requires-decomp.